apps/devhub/views.py

```diff
@@ -628,18 +624,32 @@ def upload(request, addon=None, is_standalone=False, is_listed=True,
     fu = FileUpload.from_post(filedata, filedata.name, filedata.size)
     fu.update(automated_signing=automated)
     log.info('FileUpload created: %s' % fu.pk)
-    if request.user.is_authenticated():
-        fu.user = request.amo_user
+    if user.is_authenticated():
+        fu.user = RequestUser.objects.get(pk=user.pk)
```

Comment:
I'm not sure. It looks like this is a foreign key to `UserProfile` so I'll try setting it to `user` and seeing what happens.